src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -151,9 +144,9 @@ protected void validateOrDie(DatasetVersion dsv, Boolean lenient) throws Command
      * @param ctxt
      * @throws CommandException
      */
-    protected void registerExternalIdentifier(Dataset theDataset, CommandContext ctxt, boolean retry) throws CommandException {
+    protected void registerExternalIdentifier(DvObject theDataset, CommandContext ctxt, boolean retry) throws CommandException {
         if (!theDataset.isIdentifierRegistered()) {
-            PidProvider pidProvider = PidUtil.getPidProvider(theDataset.getGlobalId().getProviderId());
+            PidProvider pidProvider = theDataset.getEffectivePidGenerator();
```

Comment:
@qqmyers I've updated this line to make it easier to test. I'm not sure if the use of the utilities is intentional and I should revert this line...

Reply:
>there are two concepts: the PidProvider associated with the object based on it's globalId, and the PidProvider that will be used if/when this DvObjectContainer is used in creating child objects (a dataset in a a collection or files in a dataset)
Thanks for the explanation, I wasn't aware of those two concepts.
>In non-move cases, they would usually be the same.
They should also be in sync after the reconciliation. 
>Whenever you create a new one, you'd use the effectiveGenerator.
Hence, I need to update the effectiveGenerator so new child objects (datafiles) us it, right? 
This is basically the response the other comment related to setting the effectiveGenerator below.